src/base/static/utils/expression/definitions/lookup.tsx

```diff
@@ -0,0 +1,129 @@
+import { Expression, IEvaluationContext } from "../expression";
```

Comment:
I'm concerned about the complexity that we have implemented to support our expressions. How do you feel about maintaining this? Is it difficult to debug?

Also, do we have tests for this code? This seems like it would have been a good candidate for TDD.

Reply:
Definitely going to add tests for the expressions. In practice I found it relatively easy to work with and debug the parser, partly because all expressions need to conform to the same interface. But so far no one else has had to write value expressions against this system, so I don't know how easy or difficult it would be for others to work with...

Reply:
Also, this parser pattern is modeled on the Mapbox style expression parser, which provided a lot of guidance in how to put this together.

Reply:
I'm not sure how you're planning to write the tests, but I would suggest using Jest. We already have it installed, and it's a pretty good testing framework.

Here is a post that seems to sum up how to use Jest for unit testing: https://wanago.io/2018/08/27/testing-javascript-tutorial-types-of-tests-of-unit-testing-with-jest/